pkg/clients/clients.go

```diff
@@ -191,30 +196,38 @@ func (dcs *DynamicClientSet) namespaced(gvk schema.GroupVersionKind) (bool, erro
 		}
 	}
 
-	return false, fmt.Errorf("failed to discover namespace info for %s", gvk)
+	return true, fmt.Errorf("failed to discover namespace info for %s", gvk)
 }
 
 func (dcs *DynamicClientSet) getServerResources() ([]*v1.APIResourceList, error) {
 	var resources []*v1.APIResourceList
 	err := retry.SleepingRetry(
 		func(i uint) error {
-			resources, err := dcs.DiscoveryClientCached.ServerPreferredResources()
+			var err error
+			resources, err = dcs.DiscoveryClientCached.ServerPreferredResources()
 			if err != nil || len(resources) == 0 {
-				return fmt.Errorf("failed to retrieve server resources")
+				return fmt.Errorf("failed to retrieve server resources: %#v", err)
 			}
 
 			return nil
 		}).
 		WithMaxRetries(5).
 		WithBackoffFactor(2).
-		Do()
+		Do(discovery.IsGroupDiscoveryFailedError, isServerCacheError)
 	if err != nil {
 		return nil, err
 	}
 
 	return resources, nil
 }
 
+func isServerCacheError(err error) bool {
+	if err == cacheddiscovery.ErrCacheEmpty || err == cacheddiscovery.ErrCacheNotFound {
```

Comment:
Can we follow up to make sure we're catching any other changes to the error API? This changing of the error semantics scares me a lot.